### DISCUSSION TITLE
Cancelling PlayerBlockPlaceEvent makes world dark

### DISCUSSION THREAD

**Discussion by GliczDev** [Minestom/Minestom#1354] [Help] [answered]
Hi, cancelling PlayerBlockPlaceEvent makes world dark, how can i fix this?
My code:
```java
globalEventHandler.addListener(PlayerBlockPlaceEvent.class, event -> {
    if (!instanceContainer.getBlock(event.getBlockPosition()).isAir()) {
        event.setCancelled(true);
    }
});
```

**Comment by GoldenCookieGIT** [comment, marked as answer]
Minestom doesn't have a light engine atm so it trusts client's prediction on the light. To fix this you would set the instance's dimension to have an ambient light of 2f
